src/main/java/fruitshop/strategy/SupplyOperation.java

```diff
@@ -0,0 +1,18 @@
+package fruitshop.strategy;
+
+import fruitshop.db.Storage;
+import fruitshop.model.FruitTransaction;
+
+public class SupplyOperation implements OperationHandler {
+    @Override
+    public void handler(FruitTransaction fruitTransaction) {
+        Storage storage = new Storage();
```

Comment:
Creating a new instance of `Storage` here means that each `SupplyOperation` will have its own `Storage`, which is not consistent with the idea of having a single storage that maintains the state of the inventory. Use a shared `Storage` instance instead.